### PR TITLE
Refactor for using stack

### PR DESCRIPTION
**Pull request by xgroleau**

### CODE REVIEW

File: src/bsp/src/stm32/src/SocketFactory.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@ std::optional<TCPClient> SocketFactory::createTCPClient(const char* address,
     // Connect the socket
     if (lwip_connect(sockfd, (sockaddr*)&serverAddr, sizeof(serverAddr)) < 0) {
         logger.log(LogLevel::Info, "Could not connect to server %s : %d", address, port);
+        lwip_close(sockfd);
```
Comment (cquesnel) [inline src/bsp/src/stm32/src/SocketFactory.cpp:26]: Do we need to do the same thing on the POSIX version?

File: src/bsp/src/posix/include/TCPUartMock.h [hunk 2]
```diff
@@ -24,6 +25,8 @@ class TCPUartMock : public IHostUart {
   private:
     ILogger& m_logger;
 
+    BaseTask<configMINIMAL_STACK_SIZE * 100> m_listenTask;
```
Comment (cquesnel) [inline src/bsp/src/posix/include/TCPUartMock.h:28]: Was the x100 a test or do we really need such a big stack?

**Review by cquesnel — APPROVED** [review]

**Review by cquesnel — APPROVED** [review]
GJ! Very clean with the tasks and especially the LockGuard

File: src/bsp/src/stm32/src/SocketFactory.cpp (hunk 1 above)
Comment (xgroleau) [reply src/bsp/src/stm32/src/SocketFactory.cpp:26]: hmmmm good question, I'll check it out

File: src/bsp/src/posix/include/TCPUartMock.h (hunk 2 above)
Comment (xgroleau) [reply src/bsp/src/posix/include/TCPUartMock.h:28]: It was the value that was there when I refactored it. I'll change it to something smaller

Comment (cquesnel) [reply src/bsp/src/posix/include/TCPUartMock.h:28]: my bad, it was probably a test from me

File: src/bsp/src/stm32/src/SocketFactory.cpp (hunk 1 above)
Comment (xgroleau) [reply src/bsp/src/stm32/src/SocketFactory.cpp:26]: Posix socket are now closed on fail

**Pull request merged**